content/5_Plat/Convex_Hull.mdx

```diff
@@ -85,22 +84,204 @@ export const problems = {
 
 <!-- [convex hull](https://en.wikipedia.org/wiki/Convex_hull_algorithms). -->
 
+
+## Introduction
+
+The **Convex Hull** is the subset of points that forms the smallest convex polygon which encloses all points in the set. To visualize this, imagine that each point is a pole. Then, imagine what happens if you were to wrap a rope around the outside of all the poles, and then pull infinitely hard, such that the connections between any two points that lie on the edge of the rope are lines. The set of points that touch the rope is the convex hull.
+
+<Spoiler title="Convex Hull Visualization">
+
+![Convex Hull Visualization](./assets/convexhull.jpg)
+
+</Spoiler>
+
 <FocusProblem problem={problems.sample[0]} />
 
 ## With Graham Scan
 
-- [cp-algo](https://cp-algorithms.com/geometry/grahams-scan-convex-hull.html)
-- [My Implementation](<https://github.com/bqi343/USACO/blob/master/Implementations/content/geometry%20(13)/Polygons/ConvexHull2.h>)
+<Resources>
+	<Resource source="Wikipedia" title="Graham Scan" url="https://en.wikipedia.org/wiki/Graham_scan" />
+	<Resource source="VisuAlgo" title="Graham Scan Visualization" url="https://visualgo.net/en/convexhull" />
+	<Resource source="UCSD" title="Original Graham Scan Paper" url="http://www.math.ucsd.edu/~ronspubs/72_10_convex_hull.pdf" />
+</Resources>
+
+<Youtube id="9rQMLpQn5xQ"/>
 
 ### Solution
 
-<IncompleteSection />
+<Resources>
+	<Resource source="Benq" title="Graham Scan Implementation" url="https://github.com/bqi343/USACO/blob/master/Implementations/content/geometry%20(13)/Polygons/ConvexHull2.h" />
+</Resources>
+
+<LanguageSection>
+
+<CPPSection>
+
+The Graham Scan algorithm works in 3 steps. First, it sorts all of the $n$ points by their counterclockwise angle around a pivot $P_0$, breaking ties by distance. This algorithm uses the leftmost (and bottommost if there is a tie), point as $P_0$. 
+
+To start the creation of the convex hull, we choose 2 points. The pivot (first point), and the second point based on our initial sorting.
+
+Then, we maintain a stack containing the points of the convex hull, such that the set of points in the stack is convex. As a result, we only have to check the candidate point against the topmost 2 points.
```

Comment:
> Then, we maintain a stack containing the points of the convex hull, such that the set of points in the stack is convex.

Should explicitly say here that for the set of points to be convex, every three consecutive points of the stack must form a counterclockwise turn.

Reply:
actually it's probably best to mention this *before*

> As a result, we only have to check the candidate point against the topmost 2 points.